src/modules/Packages.rb

```diff
@@ -2420,6 +2355,42 @@ def remote_x11_packages
 
   private
 
+    def fetch_selected(category)
+      items = Pkg.ResolvableProperties("", category, "").select { |i| i["status"] == :selected }
+      items.each_with_object([]) { |i, a| a << i["name"] }.sort
+    end
+
+    # Current package, pattern, product, patch and language selection.
+    #
+    # @return [Hash] selected packages, patterns, products,...
+    def current_proposal
+      { "packages"  => Pkg.GetPackages(:selected, false).compact.sort,
```

Comment:
Just a question: it is `compact` really needed here? I was taking a look to the `GetPackages` code and I am not sure that it is returning `nil` values.

Reply:
Not really. I have removed it.